### PR TITLE
Test Adapter uri cleanup

### PR DESCRIPTION
**Pull request by bschnurr**
- Joining project/workspace  pytest executor into one and moving it to the pytest folder
- adding specific ContainerDiscoverer uri strings for project and workspace
- adding specific executor uri strings for pytest
- removing TestMethodResolver.cs. Test Explorer will just run all tests in file with right clicking in editor and selecting "Run Test".

### CODE REVIEW

File: Python/Product/TestAdapter.Executor/Pytest/TestExecutorPytest.cs [hunk 1]
```diff
@@ -0,0 +1,171 @@
+// Python Tools for Visual Studio
```
Comment (bschnurr) [inline Python/Product/TestAdapter.Executor/Pytest/TestExecutorPytest.cs]: rename and moved TestExecutor to Pytest/TestExecutorPytest.cs

also removed all the old unittest execution code that was at the bottom

File: Python/Product/TestAdapter.Executor/TestDiscoverer.cs [hunk 2]
```diff
@@ -22,20 +22,18 @@
...
-    /// Note even though we we specify  [DefaultExecutorUri(PythonConstants.TestExecutorUriString)] we still get all .py source files
-    /// from all testcontainers.  Adding another ITestDiscover class with  [DefaultExecutorUri(PythonConstants.WSTestExecutorUriString)]
-    /// doesn't filter anything so just using one class for all types of .py sources
+    /// Note even though we we specify  [DefaultExecutorUri(PythonConstants.PytestExecutorUri)] we still get all .py source files
+    /// from all testcontainers.  
     /// </summary>
     [FileExtension(".py")]
-    [DefaultExecutorUri(PythonConstants.TestExecutorUriString)]
+    [DefaultExecutorUri(PythonConstants.PytestExecutorUriString)]
     public class PythonTestDiscoverer : ITestDiscoverer {
```
Comment (bschnurr) [inline Python/Product/TestAdapter.Executor/TestDiscoverer.cs]: todo: i'll rename this file to match the class PythonTestDiscoverer

File: Python/Product/TestAdapter/TestMethodResolver.cs [hunk 3]
```diff
@@ -1,64 +0,0 @@
...
-            get { return PythonConstants.ExecutorUri; }
-        }
-
-        [Obsolete]
-        public string GetCurrentTest(string filePath, int line, int lineCharOffset) {
-
-            var componentModel = _serviceProvider.GetService(typeof(SComponentModel)) as IComponentModel;
-            var testService = componentModel.GetService<ITestsService>();
-
-            var tests = testService.GetTestsAsEnumerableAsync().GetAwaiter().GetResult().ToList();
```
Comment (bschnurr) [inline Python/Product/TestAdapter/TestMethodResolver.cs]: we dont' want this to ship so deleting and the fallback is for Test Explorer to run all tests in the file when you right click in the editor and choose "run test"

File: Python/Product/TestAdapter.Executor/UnitTest/TestExecutorUnitTest.cs [hunk 4]
```diff
@@ -284,7 +284,7 @@ out testMethod
                 var node = doc.CreateNavigator().SelectSingleNode("/RunSettings/Python/DryRun[@value='true']");
                 return node != null;
             } catch (Exception ex) {
-                Debug.Fail(ex.ToUnhandledExceptionMessage(typeof(TestExecutor)));
+                Debug.Fail(ex.ToUnhandledExceptionMessage(typeof(TestExecutorPytest)));
```
Comment (huguesv) [inline Python/Product/TestAdapter.Executor/UnitTest/TestExecutorUnitTest.cs]: typeof(TestExecutorUnitTest)
or
GetType()

File: Python/Product/TestAdapter.Executor/UnitTest/TestExecutorUnitTest.cs [hunk 5]
```diff
@@ -300,7 +300,7 @@ out testMethod
                 var node = doc.CreateNavigator().SelectSingleNode("/RunSettings/Python/ShowConsole[@value='false']");
                 return node == null;
             } catch (Exception ex) {
-                Debug.Fail(ex.ToUnhandledExceptionMessage(typeof(TestExecutor)));
+                Debug.Fail(ex.ToUnhandledExceptionMessage(typeof(TestExecutorPytest)));
```
Comment (huguesv) [inline Python/Product/TestAdapter.Executor/UnitTest/TestExecutorUnitTest.cs]: typeof(TestExecutorUnitTest)
or
GetType()

File: Python/Product/TestAdapter/PythonStackTraceParser.cs [hunk 6]
```diff
@@ -31,7 +31,7 @@ namespace Microsoft.PythonTools.TestAdapter {
     sealed class PythonStackTraceParser : IStackTraceParser {
         public Uri ExecutorUri {
             get {
-                return PythonConstants.ExecutorUri;
+                return PythonConstants.PytestExecutorUri;
```
Comment (huguesv) [inline Python/Product/TestAdapter/PythonStackTraceParser.cs]: I'd change that for the unittest execution uri. The implementation is for that (unittest).

File: Python/Tests/TestAdapterTests/TestExecutorTests.cs [hunk 7]
```diff
@@ -592,7 +592,7 @@ public abstract class TestExecutorTests {
 
         private static List<TestCase> CreateTestCasesFromTestInfo(TestEnvironment testEnv, IEnumerable<TestInfo> expectedTests) {
             return Enumerable.Select(expectedTests, ti => {
-                var testCase = new TestCase(ti.FullyQualifiedName, new Uri(PythonConstants.TestExecutorUriString), ti.FilePath) {
+                var testCase = new TestCase(ti.FullyQualifiedName, new Uri(PythonConstants.PytestExecutorUriString), ti.FilePath) {
```
Comment (huguesv) [inline Python/Tests/TestAdapterTests/TestExecutorTests.cs]: make the uri conditional based on `testEnv.Framework`

File: Python/Tests/TestAdapterTests/TestExecutorTests.cs [hunk 8]
```diff
@@ -196,7 +196,7 @@ public abstract class TestExecutorTests {
             var testCases = CreateTestCasesFromTestInfo(testEnv, expectedTests);
             var runContext = new MockRunContext(runSettings, testCases, testEnv.ResultsFolderPath);
             var recorder = new MockTestExecutionRecorder();
-            var executor = new TestExecutor();
+            var executor = new TestExecutorPytest();
```
Comment (huguesv) [inline Python/Tests/TestAdapterTests/TestExecutorTests.cs]: Should be unittest executor (it's `RunUnittestCancel`)

File: Python/Product/TestAdapter/PythonStackTraceParser.cs (hunk 6 above)
Comment (huguesv) [reply Python/Product/TestAdapter/PythonStackTraceParser.cs]: (as far as i know we don't fill in the stack trace for pytest)

Comment (bschnurr) [reply Python/Product/TestAdapter/PythonStackTraceParser.cs]: ya i wasn't sure about this change. Unittest makes more sense

File: Python/Product/TestAdapter.Executor/UnitTest/TestExecutorUnitTest.cs [hunk 9]
```diff
@@ -519,7 +519,7 @@ sealed class TestRunner : IDisposable {
                         arguments,
                         _settings.WorkingDirectory,
                         _env,
-                        _showConsole,
+                        true,
```
Comment (huguesv) [inline Python/Product/TestAdapter.Executor/UnitTest/TestExecutorUnitTest.cs]: should undo

**Pull request merged**